lib/cypher.rb

```diff
@@ -0,0 +1,59 @@
+require './lib/shift_values'
+
+class Cypher < ShiftValues
+
+  def self.values(key, date)
+    Cypher.new.values(key, date)
+  end
+
+  def self.alphabet
+    ("a".."z").to_a << " "
+  end
+
+  def self.encode_matched_alphabet(values)
+    Hash[alphabet.zip(alphabet.rotate(values[0]))]
+  end
+
+  def self.encode_transform(input, values)
+    changed_input = ""
+    input.chars.each do |letter|
+      if alphabet.include?(letter) == false
+        changed_input = changed_input + letter
+        next
+      end
+      values = values.rotate unless changed_input.empty? == true
+      changed_input << encode_matched_alphabet(values)[letter]
+    end
+    changed_input
```

Comment:
This method transforms the string based on the shift value.
- start with an empty string (changed input)
- input.chars breaks the initial string into an array of individual letters
- for each letter we first figure out it the alphabet array contains the given letter
if not, that letter is shoveled into the empty string, this is to account for special characters
- otherwise, we rotate the shift values arrays, in order to make sure that the first value is appropriate for the letter we're on. We don't rotate if the string is empty, meaning that it's the first letter.
- then we use our changed alphabet based on the shift value to get the new letter and add it to the string